client/my-sites/checklist/wpcom-checklist/index.jsx

```diff
@@ -2,571 +2,57 @@
-					task,
-					url: '/me/get-apps',
-					dismiss: true,
-				} ) }
-				onDismiss={ this.handleTaskDismiss( task.id ) }
-				title={ translate( 'Get the WordPress app' ) }
-			/>
-		);
-	};
-
-	renderSiteLaunchedTask = ( TaskComponent, baseProps, task ) => {
-		const { translate } = this.props;
-
-		return (
-			<TaskComponent
-				{ ...baseProps }
-				bannerImageSrc="/calypso/images/stats/tasks/launch.svg"
-				buttonText={ translate( 'Launch site' ) }
-				completedTitle={ translate( 'You launched your site' ) }
-				description={ translate(
-					'Your site is private and only visible to you. Launch your site, when you are ready to make it public.'
-				) }
-				onClick={ this.handleLaunchSite( task ) }
-				onDismiss={ this.handleTaskDismiss( task.id ) }
-				title={ translate( 'Launch your site' ) }
-			/>
-		);
-	};
-}
-
-function getContactPage( posts ) {
-	return find(
-		posts,
-		post =>
-			post.type === 'page' &&
-			( some( post.metadata, { key: '_headstart_post', value: '_hs_contact_page' } ) ||
-				post.slug === 'contact' )
-	);
+		return <AsyncLoad require="./component.jsx" { ...ownProps } />;
+	}
 }
 
-const getTaskUrls = createSelector(
-	( state, siteId ) => {
-		const posts = getPostsForQuery( state, siteId, query );
+export default connect( state => {
+	const siteId = getSelectedSiteId( state );
 
-		const firstPost = find( posts, { type: 'post' } );
-		const siteSlug = getSiteSlug( state, siteId );
-		const contactPageID = get( getContactPage( posts ), 'ID', null );
-		const contactPageUrl = contactPageID && `/page/${ siteSlug }/${ contactPageID }`;
-
-		return {
-			post_published: compact( [ '/post', siteSlug, get( firstPost, [ 'ID' ] ) ] ).join( '/' ),
-			contact_page_updated: contactPageUrl,
-		};
-	},
-	( state, siteId, posts = getPostsForQuery( state, siteId, query ) ) => [ posts ]
-);
-
-export default connect(
-	state => {
-		const siteId = getSelectedSiteId( state );
-		const siteSlug = getSiteSlug( state, siteId );
-		const user = getCurrentUser( state );
-		const taskUrls = getTaskUrls( state, siteId );
-
-		return {
-			designType: getSiteOption( state, siteId, 'design_type' ),
-			isEligibleForDotcomChecklist: isEligibleForDotcomChecklist( state, siteId ),
-			isSiteSection: isSiteSection( state ),
-			siteId,
-			siteSlug,
-			taskStatuses: get( getSiteChecklist( state, siteId ), [ 'tasks' ] ),
-			taskUrls,
-			userEmail: ( user && user.email ) || '',
-			needsVerification: ! isCurrentUserEmailVerified( state ),
-			isSiteUnlaunched: isUnlaunchedSite( state, siteId ),
-		};
-	},
-	{
-		createNotice,
-		recordTracksEvent,
-		requestGuidedTour,
-		requestSiteChecklistTaskUpdate,
-		launchSite,
-	}
-)( localize( WpcomChecklist ) );
+	return {
+		designType: getSiteOption( state, siteId, 'design_type' ),
+		isEligibleForDotcomChecklist: isEligibleForDotcomChecklist( state, siteId ),
+		siteId,
+		taskStatuses: get( getSiteChecklist( state, siteId ), [ 'tasks' ] ),
+	};
```

Comment:
Instead of a bag of rather random properties, this `mapStateToProps` should return the two booleans the component acually cares about: `shouldRender` and `shouldShowNotification`. I.e., the calculation should be moved from the React component into the `connect` call.

Then the React component will get updated (and the `setNotification` side effects fired) if and only if one of these booleans changed.

Reply:
Great idea! Done in #28678.